client/post-editor/editor-drawer/index.jsx

```diff
@@ -296,26 +304,26 @@ const EditorDrawer = React.createClass( {
 
 		return (
 			<div className="editor-drawer">
-				{ site && ! this.currentPostTypeSupportsAll() && (
+				{ site && ! POST_TYPE_SUPPORTS[ type ] && (
 					<QueryPostTypes siteId={ site.ID } />
 				) }
-				{ 'page' === type
-					? this.renderPageDrawer()
-					: this.renderPostDrawer() }
+				{ this.renderTaxonomies() }
+				{ this.renderFeaturedImage() }
+				{ this.renderPageOptions() }
+				{ this.renderSharing() }
+				{ this.renderMoreOptions() }
 			</div>
 		);
 	}
 } );
 
 export default connect(
 	( state ) => {
-		const site = getSelectedSite( state );
-		if ( ! site ) {
-			return {};
-		}
+		const siteId = getSelectedSiteId( state );
```

Comment:
>Should we guard against this being falsey?

We needed the check previously since it couldn't be guaranteed that `getSelectedSite` would be an object (e.g. when a site is unset after navigating away from the page), so accessing `site.ID` could have thrown an error. With the `getSelectedSiteId`, at worst we'd receive `null` as the value, which `getPostType` is graceful in handling.